url8e6440c8aa_robdimsdale_project_euler_tgz-pJ8-com_rmd_personal_projecteuler_MainJ8/src/com/rmd/personal/projecteuler/p1to19/P017.java

```diff
@@ -0,0 +1,141 @@
+package com.rmd.personal.projecteuler.p1to19;
+
+import com.rmd.personal.projecteuler.Problem;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class P017 implements Problem {
+
+    private static final int DEFAULT_START = 1;
+    private static final int DEFAULT_END = 1000;
+
+    private int start;
+    private int end;
+
+    private Map<Integer, String> digitToWord;
+
+    public P017() {
+        this.setStart(DEFAULT_START);
+        this.setEnd(DEFAULT_END);
+
+        this.digitToWord = new HashMap<Integer, String>();
+        this.digitToWord.put(0, "");
+        this.digitToWord.put(1, "one");
+        this.digitToWord.put(2, "two");
+        this.digitToWord.put(3, "three");
+        this.digitToWord.put(4, "four");
+        this.digitToWord.put(5, "five");
+        this.digitToWord.put(6, "six");
+        this.digitToWord.put(7, "seven");
+        this.digitToWord.put(8, "eight");
+        this.digitToWord.put(9, "nine");
+    }
+
+    public int getStart() {
+        return start;
+    }
+
+    public void setStart(int start) {
+        this.start = start;
+    }
+
+    public int getEnd() {
+        return end;
+    }
+
+    public void setEnd(int end) {
+        this.end = end;
+    }
+
+    private Map<Integer, String> getDigitToWord() {
+        return this.digitToWord;
+    }
+
+    @Override
+    public String getDescription() {
+        return "If the numbers 1 to 5 are written out in words: one, two, three, four, five, "
+                + "then there are 3 + 3 + 5 + 4 + 4 = 19 letters used in total.\n\n"
+                + "If all the numbers from 1 to 1000 (one thousand) inclusive were written out in words, "
+                + "how many letters would be used?\n\n\n"
+                + "NOTE: Do not count spaces or hyphens. For example, 342 (three hundred and forty-two) contains 23 "
+                + "letters and 115 (one hundred and fifteen) contains 20 letters. "
+                + "The use of \"and\" when writing out numbers is in compliance with British usage.";
+    }
+
+    @Override
+    public String run() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.